json/src/main/java/org/apache/james/json/JsonGenericSerializer.java

```diff
@@ -107,10 +107,20 @@ public T deserialize(String value) throws IOException {
         }
     }
 
+    public T deserialize(String type, String value) throws IOException {
+        try {
+            JsonNode jsonNode = objectMapper.readTree(value);
+
+            DTOModule<T, U> dtoModule = retrieveModuleForType(type);
+            U dto = objectMapper.readValue(objectMapper.treeAsTokens(jsonNode), dtoModule.getDTOClass());
+            return dtoModule.getToDomainObjectConverter().convert(dto);
+        } catch (MismatchedInputException e) {
```

Comment:
you don't handle UnknownTypeException?

Reply:
It's a runtime exception. In case of unknown type, a runtime `UnknownTypeException` is thrown.
And in case of the json content not matching the provided type, a runtime `InvalidTypeException` is thrown.